plugins/reports/counter/CounterReportPlugin.inc.php

```diff
@@ -94,346 +207,72 @@ function setBreadcrumbs() {
-			$templateManager->display($this->getTemplatePath() . 'soaperror.tpl', 'text/xml');
-		} else {
-
-			$reportRequestNode = $tree->getChildByName($soapEnvPrefix.'Body')->getChildByName($counterPrefix.'ReportRequest');
-
-			$requestorID = $reportRequestNode->getChildByName($sushiPrefix.'Requestor')->getChildByName($sushiPrefix.'ID')->getValue();
-			$requestorName = $reportRequestNode->getChildByName($sushiPrefix.'Requestor')->getChildByName($sushiPrefix.'Name')->getValue();
-			$requestorEmail = $reportRequestNode->getChildByName($sushiPrefix.'Requestor')->getChildByName($sushiPrefix.'Email')->getValue();
-
-			$customerReferenceID = $reportRequestNode->getChildByName($sushiPrefix.'CustomerReference')->getChildByName($sushiPrefix.'ID')->getValue();
-
-			$reportName = $reportRequestNode->getChildByName($sushiPrefix.'ReportDefinition')->getAttribute('Name');
-			$reportRelease = $reportRequestNode->getChildByName($sushiPrefix.'ReportDefinition')->getAttribute('Release');
-
-			$usageDateRange = $reportRequestNode->getChildByName($sushiPrefix.'ReportDefinition')->getChildByName($sushiPrefix.'Filters')->getChildByName($sushiPrefix.'UsageDateRange');
-			$usageDateBegin = $usageDateRange->getChildByName($sushiPrefix.'Begin')->getValue();
-			$usageDateEnd = $usageDateRange->getChildByName($sushiPrefix.'End')->getValue();
-
-
-			$this->_assignTemplateCounterXML($templateManager, $usageDateBegin, $usageDateEnd, $useLegacyStats);
-
-			$templateManager->assign('requestorID', $requestorID);
-			$templateManager->assign('requestorName', $requestorName);
-			$templateManager->assign('requestorEmail', $requestorEmail);
-			$templateManager->assign('customerReferenceID', $customerReferenceID);
-			$templateManager->assign('reportName', $reportName);
-			$templateManager->assign('reportRelease', $reportRelease);
-			$templateManager->assign('usageDateBegin', $usageDateBegin);
-			$templateManager->assign('usageDateEnd', $usageDateEnd);
-
-			$templateManager->assign('templatePath', $this->getTemplatePath());
-
-			$templateManager->display($this->getTemplatePath() . 'sushixml.tpl', 'text/xml');
-		}
-	}
-
-	/**
-	* Internal function to form some of the CSV columns
-	*/
-	function _formColumns(&$cols, $entries) {
-		$allMonthsTotal = 0;
-		$currTotal = 0;
-		$htmlTotal = 0;
-		$pdfTotal = 0;
-		for ($i = 1; $i <= 12; $i++) {
-			$currTotal = 0;
-			foreach ($entries as $entry) {
-				$month = (int) substr($entry[STATISTICS_DIMENSION_MONTH], 4, 2);
-				if ($i== $month) {
-					$metric = $entry[STATISTICS_METRIC];
-					$currTotal += $metric;
-					if ($entry[STATISTICS_DIMENSION_FILE_TYPE] == STATISTICS_FILE_TYPE_HTML) {
-						$htmlTotal += $metric;
-					} else if ($entry[STATISTICS_DIMENSION_FILE_TYPE] == STATISTICS_FILE_TYPE_PDF) {
-						$pdfTotal += $metric;
+					// Legacy COUNTER Release 3
+					import('plugins.reports.counter.classes.reports.JR1_r3_0');
+					$r3jr1 = new JR1_r3_0('reports', $this->getName());
```

Comment:
Yes, this legacy JR1 (r3) report is exactly the report which this plugin used to deliver, COUNTER ready numbers in the old XML or XLS (CSV, really) format.  It also supports the double-legacy "legacy counter" report which is not COUNTER ready.